src/__tests__/test-utils-selectors.test.tsx

```diff
@@ -0,0 +1,141 @@
+              if (expression.property.type === 'Identifier' && expression.property.name === 'default') {
+                onExtract(filePath, propertyName);
+              }
+            }
+          }
+          if (path.node.object.type === 'MemberExpression') {
+            if (path.node.property.type === 'Identifier') {
+              parseNested(path.node.object, path.node.property.name);
+            } else if (path.node.property.type === 'StringLiteral') {
+              parseNested(path.node.object, path.node.property.value);
+            } else if (path.node.property.type === 'TemplateLiteral') {
+              parseNested(path.node.object, buildTemplateString(path.node.property));
+            } else {
+              throw new Error('Unhandled selectors access type.');
+            }
+          }
+        },
+      },
+    } as PluginObj;
+  }
+  const source = fs.readFileSync(file, 'utf-8');
+  transformSync(source, { babelrc: false, configFile: false, plugins: [extractor] })?.code;
+}
+
+function extractComponentSelectors(file: string, usedProperties: string[], onExtract: (selector: string) => void) {
+  function extractor(): PluginObj {
+    return {
+      visitor: {
+        ObjectProperty(path: NodePath<types.ObjectProperty>) {
+          if (path.node.key.type === 'StringLiteral' && matchProperties(usedProperties, path.node.key.value)) {
+            if (path.node.value.type === 'StringLiteral') {
+              onExtract(trimSelectorHash(path.node.value.value));
+            } else {
+              throw new Error('Unexpected selector value format');
+            }
+          }
+        },
+      },
+    } as PluginObj;
+  }
+  const source = fs.readFileSync(file, 'utf-8');
+  transformSync(source, { babelrc: false, configFile: false, plugins: [extractor] })?.code;
+}
+
+function trimSelectorHash(selector: string) {
+  const splitSelector = selector.replace('.', '').split('_');
+  if (splitSelector.length >= 5) {
+    splitSelector.splice(splitSelector.length - 2, splitSelector.length);
+    return splitSelector.join('_');
+  }
+  return selector;
+}
+
+function getComponentNameFromFilePath(filePath: string) {
+  return filePath.match(/lib\/components\/([\w-]+)/)![1];
+}
+
+// Build string literal from template string replacing arguments with wildcards ("*").
+function buildTemplateString(node: types.TemplateLiteral) {
+  let literal = '';
+  for (const element of flatten(zip(node.quasis, node.expressions))) {
+    if (!element) {
+      continue;
+    } else if (element.type === 'TemplateElement') {
+      literal += element.value.raw;
+    } else if (element.type === 'Identifier') {
+      literal += '*';
+    } else if (element.type === 'NumericLiteral') {
+      literal += element.value;
+    } else {
+      throw new Error('Unhandled template literal structure.');
+    }
+  }
+  return literal;
+}
+
+// Match property against the used properties supporting the wildcards ("*").
```

Comment:
In which condition is there a wildcard?

Reply:
Explained here: https://github.com/cloudscape-design/components/pull/1863#discussion_r1444768436